ml-agents/mlagents/trainers/agent_processor.py

```diff
@@ -49,17 +49,14 @@ def __init__(
         self.behavior_id = behavior_id
 
     def add_experiences(
-        self,
-        curr_info: BrainInfo,
-        next_info: BrainInfo,
-        take_action_outputs: ActionInfoOutputs,
+        self, curr_info: BrainInfo, previous_action: ActionInfo
     ) -> None:
         """
         Adds experiences to each agent's experience history.
         :param curr_info: current BrainInfo.
-        :param next_info: next BrainInfo.
         :param take_action_outputs: The outputs of the Policy's get_action method.
         """
+        take_action_outputs = previous_action.outputs
         if take_action_outputs:
             self.stats_reporter.add_stat(
                 "Policy/Entropy", take_action_outputs["entropy"].mean()
```

Comment:
I am not sure we should take the mean here. You are averaging across the Agents that took an action this step.

Reply:
This was the previous behavior with add_experiences in the trainer. I suppose the right way would be to add_stat per agent?
e.g.
```
for entropy in take_action_outputs["entropy"]:
            self.stats_reporter.add_stat(
                "Policy/Entropy", entropy
            )
```

Reply:
Changed it to this